module/plugins/hooks/Captcha9Kw.py

```diff
@@ -1,254 +1,309 @@
+            interactive = 1
+            isBase64 = 0
+
+        else:
             with open(task.captchaFile, 'rb') as f:
                 data = f.read()
-
-        except IOError, e:
-            self.log_error(e)
-            return
-
-        pluginname = re.search(r'_(.+?)_\d+.\w+', task.captchaFile).group(1)
-        option = {'min': 2,
-                  'max': 50,
-                  'phrase': 0,
-                  'numeric': 0,
-                  'case_sensitive': 0,
-                  'math': 0,
-                  'prio': min(max(self.config.get('prio'), 0), 10),
-                  'confirm': self.config.get('confirm'),
-                  'timeout': min(max(self.config.get('timeout'), 300), 3999),
-                  'selfsolve': self.config.get('selfsolve'),
-                  'cph': self.config.get('captchaperhour'),
-                  'cpm': self.config.get('captchapermin')}
-
-        for opt in str(self.config.get('hoster_options').split('|')):
-            details = map(str.strip, opt.split(';'))
-
-            if not details or details[0].lower() != pluginname.lower():
-                continue
-
-            for d in details:
-                hosteroption = d.split(" ")
-
-                if len(hosteroption) < 2 or not hosteroption[1].isdigit():
-                    continue
-
-                o = hosteroption[0].lower()
-                if o in option:
-                    option[o] = hosteroption[1]
-
-            break
-
-        post_data = {'apikey': self.config.get('passkey'),
-                     'prio': option['prio'],
-                     'confirm': option['confirm'],
-                     'maxtimeout': option['timeout'],
-                     'selfsolve': option['selfsolve'],
-                     'captchaperhour': option['cph'],
-                     'captchapermin': option['cpm'],
-                     'case-sensitive': option['case_sensitive'],
-                     'min_len': option['min'],
-                     'max_len': option['max'],
-                     'phrase': option['phrase'],
-                     'numeric': option['numeric'],
-                     'math': option['math'],
-                     'oldsource': pluginname,
-                     'pyload': 1,
-                     'source': "pyload",
-                     'base64': 1,
-                     'mouse': 1 if task.isPositional() else 0,
-                     'file-upload-01': base64.b64encode(data),
-                     'action': "usercaptchaupload"}
-
-        for _i in range(5):
+            data = b64encode(data)
+            pageurl = ""
+            interactive = 0
+            isBase64 = 1
+            self.log_debug("%s : %s" % (task.captchaFile, data))
+
+        if task.isPositional():
+            mouse = 1
+        else:
+            mouse = 0
+
+        regex = re.compile("_([^_]*)_\d+.\w+")
+        r = regex.search(task.captchaFile)
+        pluginname = r.group(1)
+
+        min_option = 1
```

Comment:
The following code is rather messy.
I think you might have used `hosteroptions` instead of `hosteroption` in some `elif` blocks.

In my opinion this should be changed in a way that a dictionary is used, as a dictionary is passed to `getURL` with the parameter `post` anyway. The values within the dictionary can be updated, depending on the values set within `hosteroptions_conf`.

The innermost if statement uses `hosteroption[1].isdigit()` for every condition. this could be checked one time before the if, by using `continue` if `hosteroption[1].isdigit()` is false.

`if len(hosteroptions) > 0:` is not necessary, because the for loop would not do anything anyway if `len` is zero.